index.html

```diff
@@ -2548,13 +2591,17 @@ <h3><dfn>Writing content</dfn></h3>
               </li>
             </ul>
           <li>
-            Switching on |record|'s <a>recordType</a>, pass |record| and |ndef|
-            to one of the following algorithms and return the result.
+            Switching on |record|'s <a>recordType</a>, invoke one of the
+            following algorithms and return the result.
             If the algorithm throws an exception |e|, reject |promise| with |e|
             and abort these steps.
             <dl>
               <dt>"`empty`"</dt>
               <ul>
+                <li>
+                  If |context| is `"smart-poster"`,
```

Comment:
This looks wrong that you restrict smart poster's creation to not include empty type, absolute-url type, external type and local type. Moreover, we should also check the  PAYLOAD field's size of smart-poster's action type and size type in creation algorithm.

Reply:
The NFC FOrum spec states that SP is basically a URL record with other records as metadata, either WKT or local type records.

Empty record and abs URL fall out from the primary intent of smart poster. While NFC Forum allows other records like ones containing vcards, we should not encourage all possible aberrations of "other records", when we have other mechanisms to include those records and when it is quite sure it will not interwork with other readers because it's not standard. If someone wants such a wildly more complex smart poster, why not use their own external type for that? 

Do we have interoperability tests of smart poster with absolute URL records and external records?

I can imagine local types should be allowed, and perhaps even external (though I'm not fully convinced on that). But empty and absolute URL? Why?

About checking size for action and size records, I agree. Moreover, it seems some readers expect the URL record to be the first record in the SP. I wonder should we include a clause in the spec that implementations should put the URL record first in the SP message?

Reply:
I feel when the NFC forum spec (the standard) says 'other records', it just means **ALL other records**, if we want to make it clear that `empty` and `absolute-url` are not allowed, we might want to change the NFC forum spec first, which is source of truth for WebNFC. Although WebNFC provides some more of web things, I assume it's not expected to interpret the standard in its own way.

About the external type record, actually it is able to replace all other types of records if users want.

Reply:
We don't need to change the NFC Forum spec for defining what "other records" we accept in SP. The NFC Forum spec is rather undefined in this respect, but the intent and use case for SP is clear.
Web NFC implementation is an "application" from NFC Forum pov, and the spec says that applications MAY ignore other records in the SP.
The discussion here is what to allow there in Web NFC.

It does not make sense to allow just anything, as it was explained so many times in previous comments. Could you please show me a SP tag that contains absolute URL or empty records, and tell how many SP tags contain external or even other local type records? 

BTW there is a hard restriction on SP that's not captured in the Web NFC spec: if there are other URI records on the same level as SP, they MUST be ignored and SP MUST be used (i.e. SP overrides all other URI records). 

So absolute URL records MUST NOT be accepted at least.
Empty and unknown records make no sense to accept.
External and other local are debatable, but I have no hard opinion on that. We could accept them.
Other MIME records are fine and they are the primary examples in the SP spec.

I am curious what do you @kenchris and @beaufortfrancois think.

Reply:
Also, as already said above, it's better policy to restrict now (in debatable situations) and allow later than vice versa.

Reply:
Continued in #547.